app/models/classification.rb

```diff
@@ -98,11 +98,16 @@ def ns
   # rubocop:disable Style/NumericPredicate
 
   def self.classify(obj, category_name, entry_name, is_request = true)
-    cat = Classification.lookup_by_name(category_name, obj.region_id)
-    unless cat.nil?
-      ent = cat.find_entry_by_name(entry_name, obj.region_id)
-      ent.assign_entry_to(obj, is_request) unless ent.nil? || obj.is_tagged_with?(ent.to_tag, :ns => "none")
-    end
+    cat = Classification.find_by_name(category_name, obj.region_id)
+    return " - FAILED. Tag category '#{category_name}' not found in region #{obj.region_id}" if cat.nil?
```

Comment:
Are these going to make their way to the UI?  If so they need to be I18N'd properly.

Separately, I think it's weird that this method returns failure strings.

I think better interfaces would be
- an Exception with the error and the caller catches that and presents it
- on success return `true`;  on failure return `[false, message]` where message is the actual failure message without the leading `- FAILURE`.  The UI should be responsible for checking the true/false value, then present the message with whatever prefix it needs

Reply:
@Fryguy a number of places including automate rely upon the fact that exceptions are not thrown.

Do you propose we update automate to handle a thrown exception here?

We could add these rescues in the various places and then merge the change here.

going the way you proposed is how Yuri originally did it. But we reverted because so much stuff seemed to break.

Reply:
> an Exception with the error and the caller catches that and presents it

this PR https://github.com/ManageIQ/manageiq/pull/20197 was reverted since it affect automation 


>Are these going to make their way to the UI? If so they need to be I18N'd properly.

return of method was not checked previously and made use of in  ManageIQ/manageiq-api#843. In that place returned message was not go over I18N' before.
If it should be I18N'd properly may be it should go to another PR ?

Reply:
What do you think about creating an Errors object and returning that? It a hack because you don't have an actual object to add it to but, at least it would be more model like

Something like - 
```ruby
e=ActiveModel::Errors.new(Classification.new)
e.add(:category, "'#{category_name}' not found in region #{obj.region_id")
```

Reply:
it looks slightly over-designed - create new object only to use as wrapper for error string  and on consumer side check if that object returned (and not nil) and extract that string from  object